src/Access.hs

```diff
@@ -429,8 +431,8 @@ makePrisms ''NeedCap
 makeLenses ''DelegateTo
 makePrisms ''DelegateTo
 
--- FIXME: dependent types needed here
-needCap :: {-cap :: -}Capability -> AccessCheck (NeedCap cap)
+-- (the type signature would be more interesting with dependent types.)
```

Comment:
There is a way to fake this kind of dependent type in Haskell but this might not be worth the effort.
The issue here being that: `needCap CanEdit :: AccessCheck (NeedCan CanVote)`